### PR TITLE
Migration - Access Group Services from bluemix-go library to Platform SDK

### PR DESCRIPTION
**Pull request by narinder-kaur**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/resource_ibm_iam_access_group_members.go [hunk 1]
```diff
@@ -98,9 +97,11 @@ func resourceIBMIAMAccessGroupMembersCreate(context context.Context, d *schema.R
...
 
-	request := prepareMemberAddRequest(userids, serviceids)
+	members := prepareMemberAddRequest(iamAccessGroupsClient, userids, serviceids)
 
-	_, err = iamuumClient.AccessGroupMember().Add(grpID, request)
+	addMembersToAccessGroupOptions := iamAccessGroupsClient.NewAddMembersToAccessGroupOptions(grpID)
+	addMembersToAccessGroupOptions.SetMembers(members)
+	_, _, err = iamAccessGroupsClient.AddMembersToAccessGroup(addMembersToAccessGroupOptions)
 	if err != nil {
 		return diag.FromErr(err)
```
Comment (kavya498) [inline ibm/resource_ibm_iam_access_group_members.go:106]: return diag.FromErr(err) ===> return diag.FromErr(fmt.Errorf("Error adding members to group(%s). API responce: %s", grpID, detailResponse))

File: ibm/resource_ibm_iam_access_group_members.go [hunk 2]
```diff
@@ -229,9 +233,11 @@ func resourceIBMIAMAccessGroupMembersUpdate(context context.Context, d *schema.R
...
 		}
-		request := prepareMemberAddRequest(userids, serviceids)
+		members := prepareMemberAddRequest(iamAccessGroupsClient, userids, serviceids)
 
-		_, err = iamuumClient.AccessGroupMember().Add(grpID, request)
+		addMembersToAccessGroupOptions := iamAccessGroupsClient.NewAddMembersToAccessGroupOptions(grpID)
+		addMembersToAccessGroupOptions.SetMembers(members)
+		_, _, err = iamAccessGroupsClient.AddMembersToAccessGroup(addMembersToAccessGroupOptions)
 		if err != nil {
 			return diag.FromErr(err)
```
Comment (kavya498) [inline ibm/resource_ibm_iam_access_group_members.go:242]: return diag.FromErr(err) ===> return diag.FromErr(fmt.Errorf("Error updating members to group(%s). API responce: %s", grpID, detailResponse))

File: ibm/resource_ibm_iam_access_group_members.go [hunk 3]
```diff
@@ -262,7 +269,8 @@ func resourceIBMIAMAccessGroupMembersUpdate(context context.Context, d *schema.R
 			if err != nil {
 				return diag.FromErr(fmt.Errorf("ERROR] Error Getting Service Ids %s %s", err, resp))
 			}
-			err = iamuumClient.AccessGroupMember().Remove(grpID, *serviceID.IamID)
+			removeMembersFromAccessGroupOptions := iamAccessGroupsClient.NewRemoveMemberFromAccessGroupOptions(grpID, *serviceID.IamID)
+			_, err = iamAccessGroupsClient.RemoveMemberFromAccessGroup(removeMembersFromAccessGroupOptions)
 			if err != nil {
 				return diag.FromErr(err)
```
Comment (kavya498) [inline ibm/resource_ibm_iam_access_group_members.go:275]: return diag.FromErr(err) ===> return diag.FromErr(fmt.Errorf("Error removing members to group(%s). API response: %s", grpID, detailResponse))

File: ibm/resource_ibm_iam_access_group_members.go [hunk 4]
```diff
@@ -300,7 +308,9 @@ func resourceIBMIAMAccessGroupMembersDelete(context context.Context, d *schema.R
 		if err != nil {
 			return diag.FromErr(err)
 		}
-		err = iamuumClient.AccessGroupMember().Remove(grpID, ibmUniqueID)
+
+		removeMembersFromAccessGroupOptions := iamAccessGroupsClient.NewRemoveMemberFromAccessGroupOptions(grpID, ibmUniqueID)
+		_, err = iamAccessGroupsClient.RemoveMemberFromAccessGroup(removeMembersFromAccessGroupOptions)
 		if err != nil {
 			return diag.FromErr(err)
```
Comment (kavya498) [inline ibm/resource_ibm_iam_access_group_members.go:315]: return diag.FromErr(err) ===> return diag.FromErr(fmt.Errorf("Error removing members to group(%s). API response: %s", grpID, detailResponse))

File: ibm/resource_ibm_iam_access_group_members.go [hunk 5]
```diff
@@ -314,7 +324,12 @@ func resourceIBMIAMAccessGroupMembersDelete(context context.Context, d *schema.R
...
 		}
-		err = iamuumClient.AccessGroupMember().Remove(grpID, *serviceID.IamID)
+
+		removeMembersFromAccessGroupOptions := &iamaccessgroupsv2.RemoveMemberFromAccessGroupOptions{
+			AccessGroupID: &grpID,
+			IamID:         serviceID.IamID,
+		}
+		_, err = iamAccessGroupsClient.RemoveMemberFromAccessGroup(removeMembersFromAccessGroupOptions)
 		if err != nil {
 			return diag.FromErr(err)
```
Comment (kavya498) [inline ibm/resource_ibm_iam_access_group_members.go:334]: return diag.FromErr(err) ===> return diag.FromErr(fmt.Errorf("Error removing iam ID members to group(%s). API responce: %s", grpID, detailResponse))

File: ibm/config.go [hunk 6]
```diff
@@ -2025,6 +2019,28 @@ func (c *Config) ClientSession() (interface{}, error) {
...
+			iamAccessGroupsURL = contructEndpoint("private.iam", cloudEndpoint)
+		}
+	}
+	iamAccessGroupsOptions := &iamaccessgroups.IamAccessGroupsV2Options{
+		Authenticator: authenticator,
+		URL:           envFallBack([]string{"IBMCLOUD_IAM_API_ENDPOINT"}, iamAccessGroupsURL),
+	}
+	iamAccessGroupsClient, err := iamaccessgroups.NewIamAccessGroupsV2(iamAccessGroupsOptions)
+	if err != nil {
+		session.vpcErr = fmt.Errorf("Error occured while configuring IAM Policy Management service: %q", err)
```
Comment (kavya498) [inline ibm/config.go:2037]: @narinder-kaur , It should be
session.iamAccessGroupsErr = fmt.Errorf("Error occured while configuring IAM Access Group service: %q", err)

File: ibm/config.go [hunk 7]
```diff
@@ -933,7 +934,6 @@ func (c *Config) ClientSession() (interface{}, error) {
 		session.globalTaggingConfigErr = errEmptyBluemixCredentials
 		session.globalTaggingConfigErrV1 = errEmptyBluemixCredentials
 		session.hpcsEndpointErr = errEmptyBluemixCredentials
-		session.iamUUMConfigErrV2 = errEmptyBluemixCredentials
```
[the remark this answers is not in the document]
Comment (narinder-kaur) [reply ibm/config.go:947]: @kavya498 this is not being used anywhere. Do we need to add this?

File: ibm/resource_ibm_iam_access_group_dynamic_rule.go [hunk 8]
```diff
@@ -244,15 +246,10 @@ func resourceIBMIAMDynamicRuleExists(d *schema.ResourceData, meta interface{}) (
...
-	if err != nil {
-		if apiErr, ok := err.(bmxerror.RequestFailure); ok {
-			if apiErr.StatusCode() == 404 {
-				return false, nil
-			}
-		}
-		return false, fmt.Errorf("Error communicating with the API: %s", err)
+	getAccessGroupRuleOptions := iamAccessGroupsClient.NewGetAccessGroupRuleOptions(grpID, ruleID)
+	rule, detailResponse, err := iamAccessGroupsClient.GetAccessGroupRule(getAccessGroupRuleOptions)
+	if err != nil || detailResponse.StatusCode == 404 {
```
[the remark this answers is not in the document]
Comment (narinder-kaur) [reply ibm/resource_ibm_iam_access_group_dynamic_rule.go:251]: sure Kavya. I just adjusted the conditions
```if detailResponse.StatusCode == 404 {
		d.SetId("")
		return false, nil
	}
	if err != nil {
		return false, fmt.Errorf("Error getting group(%s) rule(%s). API responce: %s", grpID, ruleID, detailResponse)
	}
```

**Pull request merged**